frontend/app/src/components/trade/send/TradeAmountInput.vue

```diff
@@ -0,0 +1,152 @@
+<script setup lang="ts">
+import AmountDisplay from '@/components/display/amount/AmountDisplay.vue';
+import AmountInput from '@/components/inputs/AmountInput.vue';
+import { useTradeAsset } from '@/composables/trade/asset';
+import { bigNumberifyFromRef } from '@/utils/bignumbers';
+import { bigNumberify } from '@rotki/common';
+import { get, set } from '@vueuse/core';
+import { computed } from 'vue';
+
+const model = defineModel<string>({ required: true });
+
+const props = defineProps<{
+  asset: string;
+  chain: string;
+  address?: string;
+  max: string;
+  amountExceeded: boolean;
+  loading?: boolean;
+}>();
+
+const { t } = useI18n();
+
+const { address, asset, chain } = toRefs(props);
+
+const fiatValue = ref<string>('0');
+const fiatValueToBigNumber = bigNumberifyFromRef(fiatValue);
+
+const amountToBigNumber = bigNumberifyFromRef(model);
+
+const isAmountSelected = ref<boolean>(true);
+
+const { getAssetDetail } = useTradeAsset(address);
+const assetDetail = getAssetDetail(asset, chain);
+
+// Computed properties for conversion
+const amountModel = computed({
+  get: () => get(model),
+  set: (value: string) => {
+    set(model, value);
+    const price = get(assetDetail)?.price;
+    if (price) {
```

Comment:
was there any issue with using watchers for this an the other?

Reply:
You're right, can be simplified